src/tools/perf/perftest_mad.c

```diff
@@ -245,20 +227,25 @@ static ucs_status_t perftest_mad_recv(perftest_mad_rte_group_t *rte_group,
     }
 
     len -= IB_VENDOR_RANGE2_DATA_OFFS;
-
-    status = perftest_mad_get_remote_port(user_mad, remote_port);
-    if (status != UCS_OK) {
-        goto out;
+    if (len < 0) {
+        if (user_mad->status == ETIMEDOUT) {
+            ucs_error("MAD: receive: remote unreachable");
+            status = UCS_ERR_UNREACHABLE;
+            goto out;
```

Comment:
IMO better move "goto out" outside of if/else

Reply:
fixed